gadget/install/install.go

```diff
@@ -393,22 +408,22 @@ func structureFromPartDevice(diskVol *gadget.OnDiskVolume, partNode string) (*ga
 	return nil, fmt.Errorf("cannot find partition %q", partNode)
 }
 
-// laidOutStructureForDiskStructure searches for the laid out structure that
-// matches a given OnDiskStructure.
-func laidOutStructureForDiskStructure(laidVols map[string]*gadget.LaidOutVolume, gadgetVolName string, onDiskStruct *gadget.OnDiskStructure) (*gadget.LaidOutStructure, error) {
+// laidOutStructureForPartLabel searches for the laid out structure that
+// matches a given partition label.
+func laidOutStructureForPartLabel(laidVols map[string]*gadget.LaidOutVolume, gadgetVolName string, partLabel string) (*gadget.LaidOutStructure, error) {
```

Comment:
I had to change this so it could be used with labels coming from gadget.VolumeStructure instead of just gadget.OnDiskStructure, so it needs to be like this for the moment. Let's see if that changes in stage 2 of the refactoring.